### PR TITLE
Private Event

### PR DESCRIPTION
**Pull request by kensayo**
### Private events - Associations
In this project, we build a private event web app where members can register attendance for an event or being able to create an event. Inside this app, members can see a list of past and upcoming events, if the customer is not already logged out they can only see the entire list, but they can't create or register attendance. You can find in this project relations among three tables to register users, events, and attendance.

In this project we consider adding the next sections:

- Users, Events, Attendances tables
- Users, Events, Attendances models
- Users, Events, Attendances controllers
- Views for Events and Users, only for some methods.
- Linters
- License
- Descriptive README.md
- Finally,  no error in linters :)

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coder, 👋🏽 
**This is a great project you have going on, well-done lad**. 

### Well-done 💯 
    1. The readme is just marvelous. 👍🏽 
    2. The app runs smoothly as expected.

**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [ ] Models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [ ] Authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [ ]  Event management features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Review by bini-i — CHANGES_REQUESTED** [review]
# Status: Changes Required :yellow_circle:            
### Hello @kensayo  :wave: 

Your project looks very good and shows you have put a lot of effort into it, the following are some of the highlights of your project:
#### To highlight
- Descriptive and well-written README and pull request description.
- Appealing UI and styling.
- Thin controllers with less logic in them.

Thanks for working on the requirements suggested by the previous reviewer and your project meets almost all the requirements, please work on the following missing requirement and submit for another review:
#### To improve
- As the previous reviewer stated, kindly include integration tests for a user login(with authentication) and event management.

> hint: Integration test is a test of how your application works as a whole rather than testing for functionality of unit parts of our application. 

> Your test needs to log in to your app and perform different tasks, in this case, create an event, attend an event ... 
you can refer to [capybara docs](https://github.com/teamcapybara/capybara#using-capybara-with-rspec) and this awesome [article](https://rubyyagi.com/intro-rspec-capybara-testing/).

#### If you have any questions feel free to ask within the PR and ping me on slack @Binyam Hailemeskel

### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (Code Reviewer)
> GitHub: [@bini-i](https://github.com/bini-i) 
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED :green_circle: :clap: 
------
Hello @kensayo, great job you did in this project :muscle: :100: good job working on the required changes :clap: 
-------------
Now you can merge this branch :1st_place_medal: :smile: 
HAPPY CODING!!! AND GOOS LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover:

**Pull request merged**
